report/id.go

```diff
@@ -21,9 +26,38 @@ const (
 	EdgeDelim = "|"
 )
 
+var (
+	idCache = gcache.New(1024).LRU().Build()
+	hashers = sync.Pool{
+		New: func() interface{} {
+			return fnv.New64a()
+		},
+	}
+)
+
+func lookupID(part1, part2, part3 string, f func() string) string {
+	h := hashers.Get().(hash.Hash64)
+	h.Write([]byte(part1))
+	h.Write([]byte(part2))
+	h.Write([]byte(part3))
+	sum := h.Sum64()
+	var result string
+	if id, err := idCache.Get(sum); id != nil && err != nil {
```

Comment:
That condition surely is wrong and results in this branch becoming unreachable; I suspect you want `id != nil && err == nil`.

Given that this bug renders the cache ineffective, any performance gains from this PR cannot be attributed to it. So some new measurements should be taken to ascertain whether the cache actually produces a noticeable gain.